src/raftstore/store/fsm/peer.rs

```diff
@@ -272,6 +271,11 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
                 PeerMsg::SignificantMsg(msg) => self.on_significant_msg(msg),
                 PeerMsg::CasualMessage(msg) => self.on_casual_msg(msg),
                 PeerMsg::Start => self.start(),
+                PeerMsg::HeartbeatPd => {
+                    if self.fsm.peer.is_leader() {
+                        self.register_pd_heartbeat_tick()
```

Comment:
Why not heartbeat pd immediately?

Reply:
To reduce heartbeats. There may be pd heartbeats already. And in the past, it also take at most 10 seconds to heartbeat the new PD.